### PR TITLE
led: move led functions to new comms api, add led command line tool

### PR DESCRIPTION
**Pull request by GravesJake**

### CODE REVIEW

File: host/greatfet/protocol/vendor_requests.py [hunk 1]
```diff
@@ -26,7 +26,6 @@
 
     # Temporary, custom stuffs?
     'ENABLE_USB1',
-    'SET_LEDS',
```
Comment (dominicgs) [inline host/greatfet/protocol/vendor_requests.py]: Don't remove this line, it changes the index of all the ones below it.

File: host/greatfet/peripherals/led.py [hunk 2]
```diff
@@ -28,38 +23,15 @@ def __init__(self, board, led_number):
...
-    def _led_request(self, request_type):
-        """ Internal function that issues a request to the given board for this LED.
-
-        request_type -- The type of request to be issued. Should be one of the 
-                LED_REQUEST_xxx properties of this class.
-        """
-        self.board.comms._vendor_request_out(vendor_requests.SET_LEDS, index=request_type, value=self.led_number)
-
-
-    def set(self, on=True):
```
Comment (dominicgs) [inline host/greatfet/peripherals/led.py]: Have you checked to see that we're not using these function calls anywhere else?

File: host/greatfet/commands/greatfet_led.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import greatfet
+from greatfet import GreatFET
+from greatfet.utils import log_silent, log_verbose
+from greatfet.peripherals.led import LED
+
+
+def main():
+    from greatfet.utils import GreatFETArgumentParser
+    # Set up a simple argument parser.
+    parser = GreatFETArgumentParser(description="Utility for I2C communication via GreatFET")
```
Comment (dominicgs) [inline host/greatfet/commands/greatfet_led.py]: This isn't an I2C tool, I think you need to update the arg parsing code here, as well as the help message.

File: host/greatfet/protocol/vendor_requests.py (hunk 1 above)
Comment (GravesJake) [reply host/greatfet/protocol/vendor_requests.py]: Fixed.

File: host/greatfet/commands/greatfet_led.py (hunk 3 above)
Comment (GravesJake) [reply host/greatfet/commands/greatfet_led.py]: Updated

**Pull request merged**